### PR TITLE
[Mellanox] Integrate HW-MGMT 7.0030.4003 Changes

### PR DESCRIPTION
**Pull request by vivekrnv**
#### Why I did it

Integrate HW-MGMT 7.0030.4003 Changes

 ## Patch List
* 0057-Documentation-ABI-Add-new-attribute-for-mlxreg-io-sy.patch : https://github.com/torvalds/linux/commit/e7210563432a
* 0087-platform_data-mlxreg-Add-capability-bit-and-mask-fie.patch :
* 0088-platform-mellanox-mlxreg-hotplug-Add-support-for-new.patch :
* 0089-platform-mellanox-mlx-platform-Change-register-name.patch :
* 0090-platform-mellanox-mlx-platform-Add-support-for-new-X.patch :
* 0093-hwmon-pmbus-Add-support-for-MPS-Multi-phase-mp2855-c.patch :

#### How I did it
Run make integrate-mlnx-hw-mgmt

### CODE REVIEW

**Review by dgsudharsan — APPROVED** [review]

File: patch/0093-hwmon-pmbus-Add-support-for-MPS-Multi-phase-mp2855-c.patch [hunk 1]
```diff
@@ -0,0 +1,457 @@
...
++++ b/drivers/hwmon/pmbus/Kconfig
+@@ -290,6 +290,16 @@ config SENSORS_MAX8688
+ 	  This driver can also be built as a module. If so, the module will
+ 	  be called max8688.
+ 
++config SENSORS_MP2855
++	tristate "MPS MP2855"
++	help
++	  If you say yes here you get hardware monitoring support for MPS
++	  MP2855 Daula-Loop, Digital, Multi-Phase Controller with PMBus
```
Comment (saiarcot895) [inline patch/0093-hwmon-pmbus-Add-support-for-MPS-Multi-phase-mp2855-c.patch:48]: Nitpicking, but possible typo in "Daula"

File: patch/kconfig-inclusions [hunk 2]
```diff
@@ -107,6 +115,12 @@ CONFIG_SENSORS_LM25066=m
 CONFIG_SENSORS_UCD9000=m
 CONFIG_SENSORS_UCD9200=m
 CONFIG_THERMAL_OF=y
+CONFIG_SPI_AMD=m
+CONFIG_PINCTRL_AMD=y
+CONFIG_EDAC_AMD64=m
+CONFIG_AMD_XGBE_DCB=y
+CONFIG_AMD_XGBE_HAVE_ECC=y
+CONFIG_USB_NET_CDCETHER=m
```
Comment (saiarcot895) [inline patch/kconfig-inclusions:123]: Are `CONFIG_USB_NET_DRIVERS`, `CONFIG_USB_USBNET`, and `CONFIG_USB_NET_CDCETHER` actually needed? The last two appear to be for older or specialized NICs.

File: patch/kconfig-inclusions [hunk 3]
```diff
@@ -107,6 +115,12 @@ CONFIG_SENSORS_LM25066=m
 CONFIG_SENSORS_UCD9000=m
 CONFIG_SENSORS_UCD9200=m
 CONFIG_THERMAL_OF=y
+CONFIG_SPI_AMD=m
+CONFIG_PINCTRL_AMD=y
+CONFIG_EDAC_AMD64=m
+CONFIG_AMD_XGBE_DCB=y
+CONFIG_AMD_XGBE_HAVE_ECC=y
```
Comment (saiarcot895) [inline patch/kconfig-inclusions:122]: This config looks like something that shouldn't be directly configured, but should rather be set by some other config.

File: patch/0093-hwmon-pmbus-Add-support-for-MPS-Multi-phase-mp2855-c.patch (hunk 1 above)
Comment (vivekrnv) [reply patch/0093-hwmon-pmbus-Add-support-for-MPS-Multi-phase-mp2855-c.patch]: Handled

File: patch/kconfig-inclusions (hunk 3 above)
Comment (vivekrnv) [reply patch/kconfig-inclusions]: removed

**Review by saiarcot895 — APPROVED** [review]

**Pull request merged**